TSPL.docc/ReferenceManual/Expressions.md

```diff
@@ -1477,6 +1477,48 @@ A single expression inside parentheses is a parenthesized expression.
 > *tuple-element-list* → *tuple-element* | *tuple-element* **`,`** *tuple-element-list* \
 > *tuple-element* → *expression* | *identifier* **`:`** *expression*
 
+### Parameter-Pack Expression
+
+XXX OUTLINE:
+
+- A `repeat` expression must contain one or more `each` expressions
+  and has the shape `repeat <#repetition pattern#>`
+
+- TODO list of contexts where expansion is supported
+
+  + in a tuple, producing tuple elements
+  + as a statement, including at top level, repeating the statement's expression
+  + but not in a function call, producing arguments
+
+- The *repetition pattern* is repeated once for each type in the pack
+
+- If an expression includes both the `repeat` operator and
+  a `try` or `await` operator,
+  the `repeat` operator must appear first.
+  (So `repeat try each foo` or `repeat each try foo`)
+
+- All of the `each` expressions in a parameter-pack expression
```

Comment:
I'm having a similar reaction to the hyphen in `parameter-pack` here, but am starting to wonder whether this is just my lack of familiarity with our formal documentation and this is some convention in certain circumstances? Because I am not seeing the hyphen used in `parameter-pack` elsewhere in this PR.

Reply:
In these cases 'parameter pack' is being used as a compound adjective, so it should be hyphenated. My favorite case is something like 'drag and drop' vs. 'drag-and-drop operation': https://support.apple.com/guide/applestyleguide/d-apsg7af4f5d0/1.0/web/1.0#:~:text=drag-and-drop

Reply:
I think what's strange here is the overall term "parameter-pack expression". We should call it the "pattern expression".

Reply:
(No opinion here either way, but there might be confusion with this other meaning of pattern: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/patterns , although 'pattern expression' isn't used in that context.)